starlark/hashtable_test.go

```diff
@@ -34,58 +34,84 @@ func BenchmarkStringHash(b *testing.B) {
 }
 
 func BenchmarkHashtable(b *testing.B) {
-	// TODO(adonovan): MakeInt probably dominates the cost of this benchmark.
-	// Optimise or avoid it.
 	for i := 0; i < b.N; i++ {
 		testHashtable(b, nil)
 	}
 }
 
+func mustInt(i Int) int {
+	v, ok := i.Int64()
+	if !ok {
+		panic("bad int")
+	}
+	return int(v)
+}
+
 // testHashtable is both a test and a benchmark of hashtable.
 // When sane != nil, it acts as a test against the semantics of Go's map.
 func testHashtable(tb testing.TB, sane map[int]bool) {
+	// Set up a stream of Ints to use.
```

Comment:
Consider making ints a global variable; it would simplify the test/benchmark logic as it wouldn't need to stop/start the timer.